### ISSUE TITLE
[Feature] Add 'Add to Collection/Wantlist' and Collection/Wantlist viewing functionality to StashDB (Optional only!)

### ISSUE THREAD

**Issue by VonTittyslappen** [stashapp/stash-box#458]
**Is your feature request related to a problem? Please describe.**
I think it would be great to give people the option of cataloguing their collections through StashDB, so having a system just like Discogs; they view a scene on StashDB and there are options 'add to collection' or 'add to wantlist' - the user can then keep track of their collection and what scenes they need, a lot of people that use Stash are hoarders and this will benefit them greatly.

NOTE: I want this to be an optional thing, as obviously there will be people who dont want StashDB to keep track of what content they have. Give people the option there if they want to use it.

**Describe the solution you'd like**
Pretty much just like Discogs (music catalog website) - on Discogs you can add releases to your collection/wantlist, only a data reference point, you can then view your collection and wantlist, and when viewing a release page there are icons to show that the item is in your collection already and an icon to show that you want the release.

Adopting this system for use on StashDB would be great too! People would definitely use it.

**Describe alternatives you've considered**
N/A

**Additional context**
I can provide screenshots of the Discogs functionality if necessary.

**Comment by peolic** [comment]
I think it's a more detailed explanation of #455?

**Comment by VonTittyslappen** [comment]
> I think it's a more detailed explanation of #455?

yeah, exactly, but my issue is more in depth than just markers :)
